### DISCUSSION TITLE
contains() doesn't find fields with null value

### DISCUSSION THREAD

**Discussion by bbalaban** [nlohmann/json#4381] [Q&A] [answered]
A sample JSON object like this can validly contain null values:
{
        "firstName": "Bellina",
        "lastName": "Cornewell",
        "company": null,
        "age": 18
    }

But the contains() method always returns false for fields of null value. How can I detect null fields in a parsed object?
doing obj["company"] will not tell me anything, because it will create the field if it doesn't already exist

**Comment by nlohmann** [comment]
j.contains("company") && j["company"].is_null()

**Reply by bbalaban** [reply]
Thanks for the quick reply, but I don't see how it works. j.contains("company") will always return false

**Reply by nlohmann** [reply]
Then please share your code.

**Reply by bbalaban** [reply]
```cpp
#include <string>
#pragma warning( disable : 4068 )   // disable GCC pragma warnings from json.hpp
#include "json.hpp"     // the Lohmann stuff
using json = nlohmann::json;

void testFunc();

void testFunc()
{
    char* j = "{\"firstName\": \"Bellina\",\"company\" : null, \"address1\" : \"158 Rusk Terrace\"}";

json parsed = json::parse(j);
    if (parsed.is_null() || parsed.is_discarded())
    {
        printf("Bad JSON");
        return;
    }

    bool found = parsed.contains("address1");  // will be true
    found = parsed.contains("company");        // will be false

    std::string s = parsed.dump(-1, ' ', false);   // output will have the null field
}
```

**Reply by nlohmann** [reply, marked as answer]
Your code is odd. `contains` returns `true` if the given key is present in the object - regardless of the value.

This code works for me:

```cpp
#include <nlohmann/json.hpp>

using json = nlohmann::json;

int main()
{
    auto j = json::parse("{\"firstName\": \"Bellina\",\"company\" : null, \"address1\" : \"158 Rusk Terrace\"}");
    assert(j.contains("company"));
    assert(j.contains("address1"));
}
```

**Reply by bbalaban** [reply]
Thanks for the help. Obviously I did something wrong.
